src/function/matrix/linspace.js

```diff
@@ -0,0 +1,159 @@
+  const bignumber = load(require('../../type/bignumber/function/bignumber'))
+  const isInteger = load(require('../utils/isInteger'))
+  const add = load(require('../arithmetic/add'))
+  const substract = load(require('../arithmetic/substract'))
+  const multiply = load(require('../arithmetic/multiply'))
+  const divide = load(require('../arithmetic/divide'))
+  
+  const ONE  = new type.BigNumber(1)
+
+  /**
+   * Create an array of linearly n equally spaced points from start to end.
+   * By default, the linspace generates a array of 100 linearly equally spaced points from start
+   * to end. The number of points can be changed by the extra parameter `n`. For `n = 1` linspace
+   * returns the end.
+   *
+   * Syntax:
+   *
+   *     math.linspace(start, end)                    // Create a linspace from start to
+   *                                                  // end with 100 points.
+   *     math.linspace(start, end [, n])              // Create a linspace from start to
+   *                                                  // end with n points.
+   *
+   * Where:
+   *
+   * - `start: number | Complex`
+   *   Start of the linspace
+   * - `end: number | Complex`
+   *   End of the linspace
+   * - `n: number`
+   *   Number of generated points. Default is 100.
+   *
+   * Examples:
+   *
+   *     math.linspace(2, 200)      // [2, 4, 6, 8, ..., 200]
+   *     math.linspace(0, 100, 11)  // [0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100]
+   *     math.linspace(10, 0, 5)    // [10, 7.5, 5, 2.5, 0]
+   *
+   * See also:
+   *
+   *     logspace, range, ones, zeros
+   *
+   * @param {*} args   Parameters describing the linspace `start`, `end`, and optional `n`.
+   * @return {Array | Matrix} linspace
+   */
+  const linspace = typed('linspace', {
+    'number, number': function (start, end) {
+      return _out(_linspace(start, end, 100))
+    },
+    'number, number, number': function (start, end, n) {
+      return _out(_linspace(start, end, n))
+    },
+    'Complex, Complex': function (start, end) {
+      return _out(_complexLinspace(start, end, 100))
+    },
+    'Complex, Complex, number': function (start, end, n) {
+      return _out(_complexLinspace(start, end, n))
+    },
+    'number, Complex': function (start, end) {
+      return _out(_complexLinspace(complex(start), end, 100))
+    },
+    'Complex, number': function (start, end) {
+      return _out(_complexLinspace(start, complex(end), 100))
+    },
+    'number, Complex, number': function (start, end, n) {
+      return _out(_complexLinspace(complex(start), end, n))
+    },
+    'Complex, number, number': function (start, end, n) {
+      return _out(_complexLinspace(start, complex(end), n))
+    },
+    'BigNumber, BigNumber': function (start, end) {
+      return _out(_bigLinspace(start, end, bignumber(100)))
+    },    
+    'BigNumber, BigNumber, BigNumber': function (start, end, n) {
+      return _out(_bigLinspace(start, end, n))
+    }
+  })


```

Comment:
I dunno what these funky `\r`'s are doing here but otherwise looks great :+1:

Reply:
I believe they came from Notepad++. I did replace all for `\s$` with `<empty>`. Or it is because I didn't put my EOL to Unix style.

Reply:
Yep, those ^M's are definitely Windows line endings. On linux, use `dos2unix`. With notepad++, I believe there's a function to convert line endings to Unix.